File: src/main/java/com/zippeykeys/praisebe/block/EBuildingBlock.java
package com.zippeykeys.praisebe.block;

import com.zippeykeys.praisebe.iface.ILocalize;

import org.jetbrains.annotations.Contract;

import lombok.AllArgsConstructor;
import lombok.Getter;


@AllArgsConstructor
public enum EBuildingBlock implements ILocalize{
    MARBLE("marble"),
    BASALT("basalt");

    @Getter(onMethod_ = @Override)
    private String name;

    @Override
    @Contract(pure = true)
    public String getPrefix(){
        return "building";
    }
}

File: src/main/kotlin/com/zippeykeys/praisebe/block/EBuildingBlock.kt
package com.zippeykeys.praisebe.block

import com.zippeykeys.praisebe.iface.ILocalize

enum class EBuildingBlock(private val actualName: String) : ILocalize {
    MARBLE("marble"),
    BASALT("basalt");

    override val prefix: String
        get() = "building"

    override fun getName(): String {
        return actualName
    }
}